Makefile

```diff
@@ -29,76 +29,73 @@ jenkins: foc jenkins_build_dir
 # ================================================================
 # Genode toolchain. Only needs to be done once per target (x86/arm).
 toolchain:
-	wget https://nextcloud.os.in.tum.de/s/9idiw8BLbuwp35z/download -O toolchain
-	tar xfj toolchain -C .
+	wget -qO- https://nextcloud.os.in.tum.de/s/9idiw8BLbuwp35z/download | tar xj -C .
 #
 # ================================================================
 
 
 # ================================================================
 # Download Genode external sources. Only needs to be done once per system.
-ports: foc libports dde_linux
-
-foc:
-	./genode/tool/ports/prepare_port focnados
-
-libports:
-	./genode/tool/ports/prepare_port libc
-	./genode/tool/ports/prepare_port lwip
-	./genode/tool/ports/prepare_port stdcxx
-
-dde_linux:
-	./genode/tool/ports/prepare_port dde_linux
+ports:
+	./genode/tool/ports/prepare_port -j4 focnados libc lwip stdcxx dde_linux
 #
 # ================================================================
 
 
 # ================================================================
 # Genode build process. Rebuild subtargets as needed.
 
+CUSTOM_REPOS = $(wildcard genode-*) $(wildcard ../genode-*) toolchain-host
+
 build_dir:
+#	Create build directory
 	genode/tool/create_builddir $(GENODE_TARGET) BUILD_DIR=$(VAGRANT_GENODE_BUILD_DIR)
-	printf 'REPOSITORIES += $$(GENODE_DIR)/repos/libports\n' >> $(VAGRANT_BUILD_CONF)
-	printf 'REPOSITORIES += $$(GENODE_DIR)/../genode-dom0-HW\n' >> $(VAGRANT_BUILD_CONF)
-	printf 'REPOSITORIES += $$(GENODE_DIR)/../genode-Taskloader\n' >> $(VAGRANT_BUILD_CONF)
-	printf 'REPOSITORIES += $$(GENODE_DIR)/../genode-Parser\n' >> $(VAGRANT_BUILD_CONF)
-	printf 'REPOSITORIES += $$(GENODE_DIR)/../genode-Monitoring\n' >> $(VAGRANT_BUILD_CONF)
-	printf 'REPOSITORIES += $$(GENODE_DIR)/../genode-schedulerTest\n' >> $(VAGRANT_BUILD_CONF)
-	printf 'REPOSITORIES += $$(GENODE_DIR)/../genode-AdmCtrl\n' >> $(VAGRANT_BUILD_CONF)
-	printf 'REPOSITORIES += $$(GENODE_DIR)/../genode-Synchronization\n' >> $(VAGRANT_BUILD_CONF)
-	printf 'REPOSITORIES += $$(GENODE_DIR)/../genode-Utilization\n' >> $(VAGRANT_BUILD_CONF)
-	printf 'REPOSITORIES += $$(GENODE_DIR)/../toolchain-host\n' >> $(VAGRANT_BUILD_CONF)
-	printf 'REPOSITORIES += $$(GENODE_DIR)/../genode-CheckpointRestore-SharedMemory\n' >> $(VAGRANT_BUILD_CONF)
-	printf 'REPOSITORIES += $$(GENODE_DIR)/repos/dde_linux\n' >> $(VAGRANT_BUILD_CONF)
-	printf 'MAKE += -j4' >> $(VAGRANT_BUILD_CONF)
+
+#	Uncomment libports and dde_linux from etc/build.conf
+	for repo in libports dde_linux ; do \
+		sed -i "/$$repo/s/^#REPOSITORIES/REPOSITORIES/g" $(VAGRANT_BUILD_CONF) ; \
+	done
+
+#	Add our custom repositories to etc/build.conf
+	for repo in $(CUSTOM_REPOS); do \
+		echo "REPOSITORIES += \$$(GENODE_DIR)/../$$repo" >> $(VAGRANT_BUILD_CONF) ; \
+	done
+
+#	Speedup of the build process
+	echo "MAKE += -j4" >> $(VAGRANT_BUILD_CONF)
```

Comment:
could be only -j as this is fast if building is done with jenkins and might not be that bad for more constrained hardware.
One could also just bring it up when the make command is actually revoked

Reply:
Maybe we should stay with a sane default of 4 and change it in the jenkins build process (via a script?) since the jenkins_build_dir command is used by many for normal development on Genode and -j tends to freeze low/mid end systems.